### DISCUSSION TITLE
The best/performant way to subscribe to each fields whenever them changes

### DISCUSSION THREAD

**Discussion by Hud97** [react-hook-form/react-hook-form#9208] [Q&A]
Hi my beloved communities,

I have few tabs and each tab contain fields that I need to persist on every changes so I can store it in somewhere. I use watch api to subscribe to each changes and store it in zustand, but i can feel my apps a little bit janky/laggy. 

Any thoughts on this ?

**Comment by bluebill1049** [comment]
Try this https://react-hook-form.com/api/usewatch so re-render will be isolated, also worth seeing the bottleneck with react dev tool.

**Reply by Hud97** [reply]
alright I'll give it a try

**Comment by Moshyfawn** [comment]
If you don't care to save "real-time" form values as user typing, you can call your "save to `zustand` store" function when the user navigates to a different form tab instead, somewhere `onClick`, for example. This way, you don't to continuously update your store on each keystroke / select etc.

**Reply by Hud97** [reply]
yea, good suggestion 😁

**Reply by treywhite21** [reply]
Is there a way to make this work if the user navigates to a different step via url?

**Reply by Hud97** [reply]
> Is there a way to make this work if the user navigates to a different step via url?

Im storing it in zustand to persist my data between the url.
